### DISCUSSION TITLE
[Enhancement] [upload] 【el-upload】 How to intercept the click event of el-upload so that it authenticates before selecting the file?

### DISCUSSION THREAD

**Discussion by lu606** [element-plus/element-plus#17632] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `upload`

## Description

1. Capture the click event of el-upload
2. set some pre-customized logic to control whether the file selector pops up.

### Additional comments

_(empty)_

**Comment by warmthsea** [comment]
You can use `e.stopPropagation()`.

[demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdXBsb2FkXG4gIFxuICAgIHYtbW9kZWw6ZmlsZS1saXN0PVwiZmlsZUxpc3RcIlxuICAgIGNsYXNzPVwidXBsb2FkLWRlbW9cIlxuICAgIGFjdGlvbj1cImh0dHBzOi8vcnVuLm1vY2t5LmlvL3YzLzlkMDU5YmY5LTQ2NjAtNDVmMi05MjVkLWNlODBhZDZjNGQxNVwiXG4gICAgbXVsdGlwbGVcbiAgICA6b24tcHJldmlldz1cImhhbmRsZVByZXZpZXdcIlxuICAgIDpvbi1yZW1vdmU9XCJoYW5kbGVSZW1vdmVcIlxuICAgIDpiZWZvcmUtcmVtb3ZlPVwiYmVmb3JlUmVtb3ZlXCJcbiAgICA6bGltaXQ9XCIzXCJcbiAgICA6b24tZXhjZWVkPVwiaGFuZGxlRXhjZWVkXCJcbiAgPlxuICAgIDxlbC1idXR0b24gQGNsaWNrPVwidGVzdEZuXCIgdHlwZT1cInByaW1hcnlcIj5DbGljayB0byB1cGxvYWQ8L2VsLWJ1dHRvbj5cbiAgICA8dGVtcGxhdGUgI3RpcD5cbiAgICAgIDxkaXYgY2xhc3M9XCJlbC11cGxvYWRfX3RpcFwiPlxuICAgICAgICBqcGcvcG5nIGZpbGVzIHdpdGggYSBzaXplIGxlc3MgdGhhbiA1MDBLQi5cbiAgICAgIDwvZGl2PlxuICAgIDwvdGVtcGxhdGU+XG4gIDwvZWwtdXBsb2FkPlxuPC90ZW1wbGF0ZT5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgeyBFbE1lc3NhZ2UsIEVsTWVzc2FnZUJveCB9IGZyb20gJ2VsZW1lbnQtcGx1cydcblxuaW1wb3J0IHR5cGUgeyBVcGxvYWRQcm9wcywgVXBsb2FkVXNlckZpbGUgfSBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmNvbnN0IGZpbGVMaXN0ID0gcmVmPFVwbG9hZFVzZXJGaWxlW10+KFtcbiAge1xuICAgIG5hbWU6ICdlbGVtZW50LXBsdXMtbG9nby5zdmcnLFxuICAgIHVybDogJ2h0dHBzOi8vZWxlbWVudC1wbHVzLm9yZy9pbWFnZXMvZWxlbWVudC1wbHVzLWxvZ28uc3ZnJyxcbiAgfSxcbiAge1xuICAgIG5hbWU6ICdlbGVtZW50LXBsdXMtbG9nbzIuc3ZnJyxcbiAgICB1cmw6ICdodHRwczovL2VsZW1lbnQtcGx1cy5vcmcvaW1hZ2VzL2VsZW1lbnQtcGx1cy1sb2dvLnN2ZycsXG4gIH0sXG5dKVxuXG5jb25zdCBoYW5kbGVSZW1vdmU6IFVwbG9hZFByb3BzWydvblJlbW92ZSddID0gKGZpbGUsIHVwbG9hZEZpbGVzKSA9PiB7XG4gIGNvbnNvbGUubG9nKGZpbGUsIHVwbG9hZEZpbGVzKVxufVxuXG5jb25zdCBoYW5kbGVQcmV2aWV3OiBVcGxvYWRQcm9wc1snb25QcmV2aWV3J10gPSAodXBsb2FkRmlsZSkgPT4ge1xuICBjb25zb2xlLmxvZyh1cGxvYWRGaWxlKVxufVxuXG5jb25zdCBoYW5kbGVFeGNlZWQ6IFVwbG9hZFByb3BzWydvbkV4Y2VlZCddID0gKGZpbGVzLCB1cGxvYWRGaWxlcykgPT4ge1xuICBFbE1lc3NhZ2Uud2FybmluZyhcbiAgICBgVGhlIGxpbWl0IGlzIDMsIHlvdSBzZWxlY3RlZCAke2ZpbGVzLmxlbmd0aH0gZmlsZXMgdGhpcyB0aW1lLCBhZGQgdXAgdG8gJHtcbiAgICAgIGZpbGVzLmxlbmd0aCArIHVwbG9hZEZpbGVzLmxlbmd0aFxuICAgIH0gdG90YWxseWBcbiAgKVxufVxuXG5jb25zdCBiZWZvcmVSZW1vdmU6IFVwbG9hZFByb3BzWydiZWZvcmVSZW1vdmUnXSA9ICh1cGxvYWRGaWxlLCB1cGxvYWRGaWxlcykgPT4ge1xuICByZXR1cm4gRWxNZXNzYWdlQm94LmNvbmZpcm0oXG4gICAgYENhbmNlbCB0aGUgdHJhbnNmZXIgb2YgJHt1cGxvYWRGaWxlLm5hbWV9ID9gXG4gICkudGhlbihcbiAgICAoKSA9PiB0cnVlLFxuICAgICgpID0+IGZhbHNlXG4gIClcbn1cblxuY29uc3QgdGVzdEZuID0gKGUpPT57XG4gIGNvbnNvbGUubG9nKGUsNjY2KVxuICBlLnN0b3BQcm9wYWdhdGlvbigpXG59XG48L3NjcmlwdD5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9kaXN0L2luZGV4LmNzcycsICdodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvdGhlbWUtY2hhbGsvZGFyay9jc3MtdmFycy5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbChzdHlsZXMpXG59XG4iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvZGlzdC9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiXG4gIH0sXG4gIFwic2NvcGVzXCI6IHt9XG59IiwiX28iOnt9fQ==)
